### DISCUSSION TITLE
Installing Extensions in RStudio

### DISCUSSION THREAD

**Discussion by axiomtutor** [quarto-dev/quarto-cli#7851] [Q&A] [answered]
### Description

I'm trying to install extensions for Quarto presentations (for example, the `illinois-revealjs` extension) inside of RStudio.  My computer runs Windows 11.  

I tried using the documentation https://quarto.org/docs/extensions/managing.html

and https://quarto.org/docs/extensions/revealjs.html

and also following the instructions from the extension's GitHub.  https://github.com/coatless-quarto/illinois-revealjs

I also tried looking for any info on RStudio and its GitHub page but struck out there -- I think someone over there actually asked the same question a while ago and got no responses.  

---

After trying all of the above, I *think* it successfully downloaded the extension, but when I try to run 

```
---
title: "title"
author: "me"
format: 
  illinois-revealjs:
    chalkboard: true
editor: visual
---

## TEST
```

I get the error message

```
ERROR: Unable to read the extension 'illinois'.
Please ensure that you provided the correct id and that the extension is installed.

Stack trace:
Please ensure that you provided the correct id and that the extension is installed.
    at loadExtension (file:///C:/Users/username/AppData/Local/Programs/Quarto/bin/quarto.js:42097:15)
    ...
```

and the `...` just covers up a bunch of other very similar lines.  

However, when I set `illinois-revealjs` to just `revealjs` then it renders.

Anyone have guidance on how to do the installation / use inside of RStudio? Thanks! :)

**Comment by cderv** [comment]
> After trying all of the above, I think it successfully downloaded the extension

If the extension installed in the same folder than your document, or at the root of the Quarto project ? 

Can you check this ? 

You can run in your working directory / quarto project or document folder the following
````powershell
quarto list extensions
````

It should find the extension. 

If not installed in the right place, quarto won't find it.

**Reply by axiomtutor** [reply]
Hi, thanks for the help! 

When I run `quarto list extensions` it does print the extension.  So it seems like it's installed.

However, when I run `$ quarto update coatless/quarto-illinois` or `$ quarto add coatless/quarto-illinois` it responds with the error

```
Quarto extensions may execute code when documents are rendered. If you do not
trust the authors of the extension, we recommend that you do not install or
use the extension.
 ? Do you trust the authors of this extension (Y/n) » Yes
[>] Downloading
[>] Unzipping
ERROR: NotFound: The system cannot find the path specified. (os error 3): readdir 'C:\Users\username\AppData\Local\Temp\quarto-sessioncd50f2ec\4bfb0ae2\158ff9d5\archive\quarto-illinois-main\_extensions'

Stack trace:
    at Object.readDirSync (ext:deno_fs/30_fs.js:149:14)
    at readExtensions (file:///C:/Users/username/AppData/Local/Programs/Quarto/bin/quarto.js:42150:32)
    ...
```

and so on with more lines of the error message.  I used the instructions here:  https://github.com/coatless-quarto/illinois-revealjs

and so I ran `quarto use template coatless/quarto-illinois`.  This caused no error messages, but it also does not fix any of the previous errors.  It also does not create an example `.qmd` file as the README says.

**Reply by cderv** [reply]
I believe this is just an issue because the repo was renamed, but not the instruction in the README. 

Try 

````powershell
quarto add coatless-quarto/illinois-revealjs
````

**Reply by cderv** [reply, marked as answer]
I opened an issue so that we try to handle such case with more helpful error

Thanks for the report ! 

* https://github.com/quarto-dev/quarto-cli/issues/7886

**Reply by axiomtutor** [reply]
It worked!  Thank you one and all!
